docs/spec/sources.md

```diff
@@ -0,0 +1,342 @@
+            description:
+              "Reference to an object that will resolve to a domain name to use
+              as the sink."
+          ceOverrides:
+            type: object
+            description:
+              "Defines overrides to control modifications of the event sent to
+              the sink."
+            properties:
+              extensions:
+                type: object
+                description:
+                  "Extensions specify what attribute are added or overridden on
+                  the outbound event. Each `Extensions` key-value pair are set
+                  on the event as an attribute extension independently."
+```
+
+<!--
+
+TODO: There is still debate about how to implement the registry. Commenting out for now.
+
+### Event Type Registry
+
+The Event Type Registry is a feature that allows for software to find all CRDs
+labeled as a `Source` ducktype, and extract the defined
+[EventType](https://godoc.org/knative.dev/eventing/pkg/apis/eventing/v1alpha1#EventType)s
+that could be emitted from a source instance.
+
+To be included in the event type registry a fake `registry` component is defined
+in the `openAPIV3Schema` section inside the CRD. It is not our intention to have
+cluster operators create resources with this filled out, it is to signal to
+other software some metadata about the runtime/dataplane characteristics of this
+type of source.
+
+As an example, here is a snip the registry entry for ApiServerSource, a source
+that emits Kubernetes event,
+
+```yaml
+  validation:
+    openAPIV3Schema:
+      properties:
+        registry:
+          type: object
+          description: "Internal information, users should not set this property"
+          properties:
+            eventTypes:
+              type: object
+              description: "Event types that ApiServerSource can produce"
+              properties:
+                addResource:
+                  type: object
+                  properties:
+                    type:
+                      type: string
+                      pattern: "dev.knative.apiserver.resource.add"
+                    schema:
+                      type: string
+                deleteResource:
+                  type: object
+                  properties:
+                    type:
+                      type: string
+                      pattern: "dev.knative.apiserver.resource.delete"
+                    schema:
+                      type: string
+                updateResource:
+                  type: object
+                  properties:
+                    type:
+                      type: string
+                      pattern: "dev.knative.apiserver.resource.update"
+                    schema:
+                      type: string
+                ...
+```
+-->
+
+## Source RBAC
+
+Sources can by any shape and are expected to be extensions onto Kubernetes. To
+prevent cluster operators from re-deploying, or re-creating RBAC for all
```

Comment:
I don't understand the "re-deploying, or re-creating"

Do we mean duplicating?